automod/automod_bot.go

```diff
@@ -278,6 +280,26 @@ func (p *Plugin) handleAutomodExecution(evt *eventsystem.EventData) {
 		return
 	}
 
+	redisKey := fmt.Sprintf("automodv2_rule_execution_%d", eventData.MessageID)
+
+	var exists string
+
+	if err := common.RedisPool.Do(radix.Cmd(&exists, "GET", redisKey)); err != nil {
+		return
+	}
+	if exists == "1" {
+		return
+	}
+
+	if err := common.RedisPool.Do(radix.Cmd(nil, "SET", redisKey, "1")); err != nil {
```

Comment:
you can use SETEX instead of doing EXPIRE in a another command. 
https://redis.io/commands/setex/

Reply:
Ah, didn't know, I'll change that real quick.